### PR TITLE
crypto unit tests coverage

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: registry/endpoint_test.go [hunk 1]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
 	}
 }
 
+func TestNewV1EndPointFromStr(t *testing.T){
```
Comment (usmanniqbal) [inline registry/endpoint_test.go]: Is this test really required?

File: registry/endpoint_test.go [hunk 2]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
...
+	e, err := newV1EndpointFromStr("https://index.docker.io/v1/", nil, "", nil)
+	require.NotNil(t, e)
+	require.Nil(t, err)
+	require.EqualValues(t,e.URL.Host,"index.docker.io")
+}
+
+func TestNewV1Endpoint(t *testing.T)  {
```
Comment (usmanniqbal) [inline registry/endpoint_test.go]: Same as above, and seems that its only constructing the struct.

File: registry/endpoint_test.go [hunk 3]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
 	}
 }
 
+func TestNewV1EndPointFromStr(t *testing.T){
+	e, err := newV1EndpointFromStr("https://index.docker.io/v1/", nil, "", nil)
```
Comment (uzairalikhan) [inline registry/endpoint_test.go]: Use proper naming for variables, both (e, err) sound like error.

File: registry/endpoint_test.go [hunk 4]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
...
+	require.NotNil(t, e)
```
Comment (uzairalikhan) [inline registry/endpoint_test.go]: This function `newV1EndpointFromStr` will get tested in `TestEndpointParse` test, so no need to write its test unless you are passing it configurations different from what already used in `TestEndpointParse`, which i don't think is the case.

File: registry/endpoint_test.go [hunk 5]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
...
+	require.Nil(t, err)
```
Comment (uzairalikhan) [inline registry/endpoint_test.go]: Use assert.NoError(t, err) for checking errors.

File: registry/registry_test.go [hunk 6]
```diff
@@ -893,6 +899,34 @@ func TestIsSecureIndex(t *testing.T) {
...
+func TestTLSConfig(t *testing.T){
+	var index = makeIndex("/v1/")
+        tlsConfigDefault := tlsconfig.ServerDefault()
+	resp, err := newTLSConfig(index.Name,index.Secure)
+	require.Nil(t,err)
+	require.NotNil(t,resp)
+	require.EqualValues(t,tlsConfigDefault.ServerName,resp.ServerName)
+}
+
+func TestReadCertsDirectory(t *testing.T)  {
```
Comment (usmanniqbal) [inline registry/registry_test.go]: I dont think so we need to write tests for reading files only, and its redundant too as newTLSConfig also using it internally.

File: registry/endpoint_test.go [hunk 7]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
...
+	require.EqualValues(t,e.URL.Host,"index.docker.io")
+}
+
+func TestNewV1Endpoint(t *testing.T)  {
+	uri, err := url.Parse("http://0.0.0.0:5000/v2/")
+	require.Nil(t,err)
```
Comment (uzairalikhan) [inline registry/endpoint_test.go]: Use assert.NoError(t, err) for checking errors.

File: registry/endpoint_test.go [hunk 8]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
...
+	var index = makeIndex("/v1/")
+	tlsConfig, err := newTLSConfig(index.Name, index.Secure)
+	require.Nil(t,err)
+	e, err := newV1Endpoint(*uri,tlsConfig,"",nil)
```
Comment (uzairalikhan) [inline registry/endpoint_test.go]: same as mentioned above, variable renaming issue.

File: registry/endpoint_test.go [hunk 9]
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
...
+	require.Nil(t, err)
+	require.EqualValues(t,e.URL.Host,"index.docker.io")
+}
+
+func TestNewV1Endpoint(t *testing.T)  {
+	uri, err := url.Parse("http://0.0.0.0:5000/v2/")
+	require.Nil(t,err)
+	var index = makeIndex("/v1/")
+	tlsConfig, err := newTLSConfig(index.Name, index.Secure)
+	require.Nil(t,err)
```
Comment (uzairalikhan) [inline registry/endpoint_test.go]: same as mentioned above for checking errors.

File: registry/registry_test.go [hunk 10]
```diff
@@ -893,6 +899,34 @@ func TestIsSecureIndex(t *testing.T) {
...
+	var index =  makeIndex("/v1/")
+	hostDir := filepath.Join(CertsDir, cleanPath(index.Name))
+	tlsConfig := tlsconfig.ServerDefault()
+	err := ReadCertsDirectory(tlsConfig, hostDir)
+	require.Nil(t,err)
+}
+
+func TestNewTransport(t *testing.T) {
+	var index = makeIndex("/v1/")
+	resp, err := newTLSConfig(index.Name,index.Secure)
```
Comment (uzairalikhan) [inline registry/registry_test.go]: Use tlsConfig rather than resp.

File: registry/registry_test.go [hunk 11]
```diff
@@ -893,6 +899,34 @@ func TestIsSecureIndex(t *testing.T) {
 	}
 }
 
+
+func TestTLSConfig(t *testing.T){
+	var index = makeIndex("/v1/")
```
Comment (uzairalikhan) [inline registry/registry_test.go]: Index variable is used many times and always re declared, its better to  initialize it at the top then use it in all the tests.

File: registry/registry_test.go [hunk 12]
```diff
@@ -14,6 +14,12 @@ import (
 	"github.com/docker/distribution/registry/client/transport"
 	"github.com/docker/docker/api/types"
 	registrytypes "github.com/docker/docker/api/types/registry"
+
+	"github.com/stretchr/testify/require"
+	"github.com/docker/go-connections/tlsconfig"
+
+	"path/filepath"
+	"os"
```
Comment (uzairalikhan) [inline registry/registry_test.go]: Use proper ordering for imports and always run `gofmt` on your code before making any PR.

**Pull request closed without merging**